lectures/about_py.md

```diff
@@ -31,145 +31,354 @@ into R." -- Chris Wiggins
+* [interpreted](https://en.wikipedia.org/wiki/Interpreter_(computing)) rather than [compiled](https://en.wikipedia.org/wiki/Compiler) ahead of time.
+
+
 
 ### Syntax and Design
 
 ```{index} single: Python; syntax and design
 ```
 
-One nice feature of Python is its elegant syntax --- we'll see many examples later on.
+One reason for Python's popularity is its simple and elegant design --- we'll see many examples later on.
+
+To get a feeling for this, let's look at an example.
+
+The code below is written in [Java](https://en.wikipedia.org/wiki/Java_(programming_language)) rather than Python.
+
+You do **not** need to read and understand this code!
+
+
+```{code-block} java
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+
+public class CSVReader {
+    public static void main(String[] args) {
+        String filePath = "data.csv"; 
+        String line;
+        String splitBy = ",";
+        int columnIndex = 1; 
+        double sum = 0;
+        int count = 0;
+
+        try (BufferedReader br = new BufferedReader(new FileReader(filePath))) {
+            while ((line = br.readLine()) != null) {
+                String[] values = line.split(splitBy);
+                if (values.length > columnIndex) {
+                    try {
+                        double value = Double.parseDouble(
+                            values[columnIndex]
+                        );
+                        sum += value;
+                        count++;
+                    } catch (NumberFormatException e) {
+                        System.out.println(
+                            "Skipping non-numeric value: " + 
+                            values[columnIndex]
+                        );
+                    }
+                }
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        if (count > 0) {
+            double average = sum / count;
+            System.out.println(
+                "Average of the second column: " + average
+            );
+        } else {
+            System.out.println(
+                "No valid numeric data found in the second column."
+            );
+        }
+    }
+}
+
+```
+
+This Java code opens an imaginary file called `data.csv` and computes the mean
+of the values in the second column.
+
+Even without knowing Java, you can see that the program is long and complex.
+
+Here's Python code that does the same thing.
+
+Even if you don't yet know Python, you can see that the code is simpler and
+easier to read.
+
+```{code-cell} python3
+:tags: [skip-execution]
+
+import csv
+
+total, count = 0, 0
+with open(data.csv, mode='r') as file:
+    reader = csv.reader(file)
+    for row in reader:
+        try:
+            total += float(row[1])
+            count += 1
+        except (ValueError, IndexError):
+            pass
+print(f"Average: {total / count if count else 'No valid data'}")
+
+```
+
```

Comment:
@jstac I assume you want to keep this comparison to be pure python? This might be a nice addition. 

---

And using the python library [pandas](https://pandas.pydata.org/)

```python
import pandas as pd

data = pd.read_csv('data.csv')
data.iloc[:, 1].mean()
```

Reply:
@jstac I will merge this PR and make note of this suggestion. We can add it if you think it is a good idea.